torch/_inductor/codegen/triton.py

```diff
@@ -471,10 +471,12 @@ def __init__(self, name, bounds: ValueRanges[Any]):
 
     def update_on_args(self, name, args, kwargs):
         # When making a variable that is going to be used in indirect indexing
-        # if a where clause is used it should mean that the result is always a
+        # if we are within a masked block, it means that the result is always a
         # valid index, so you shouldn't include any of the dependent variables
         # in the resulting load mask
-        if name == "where":
+        if (
+            fx_node := getattr(V.interpreter, "current_node", None)
+        ) and fx_node.target.startswith("masked"):
```

Comment:
How does this interact with the masked sub-block? Is current_node only masked for the output of `ops.masked`?

Also pre-existing, but why do we exclude `xmask`? Shouldn't this only be filtering out `tmp{n}` masks?

Reply:
It did interact alright, but now the implementation is even cleaner without using all those global objects.